pkg/agents/team7/agent1/treaties.go

```diff
@@ -0,0 +1,173 @@
+package team7agent1
+
+import (
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+	"github.com/SOMAS2021/SOMAS2021/pkg/messages"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/health"
+)
+
+func (a *CustomAgent7) HandleBadTreaty(msg messages.ProposeTreatyMessage) {
+	reply := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), false)
+	a.SendMessage(reply)
+	a.Log("Rejected treaty", infra.Fields{"proposerID": msg.SenderID(), "proposerFloor": msg.SenderFloor(), "treatyID": msg.TreatyID()})
+}
+
+func (a *CustomAgent7) HandleProposeTreaty(msg messages.ProposeTreatyMessage) {
+	treaty := msg.Treaty()
+	// switch{
+	// case (treaty.ConditionOp() == messages.LE || treaty.ConditionOp() == messages.LT) && treaty.Condition() == messages.HP || treaty.Condition() == messages.AvailableFood :
+	// 	a.HandleBadTreaty(msg)
+	// 	return
+	// case (treaty.ConditionOp() == messages.GE || treaty.ConditionOp() == messages.GT) && treaty.Condition() == messages.Floor:
+	// 	a.HandleBadTreaty(msg)
+	// 	return
+	// case treaty.Request() == messages.Inform:
+	// 	a.HandleBadTreaty(msg)
+	// 	return
+	// case treaty.Request() == messages.LeaveAmountFood || treaty.Request() == messages.LeavePercentFood && treaty.RequestOp() == messages.LT) || treaty.RequestOp() == messages.LE:
+	// 	a.HandleBadTreaty(msg)
+	// 	return
+	// case treaty.Condition() == messages.HP && treaty.ConditionValue() < a.HealthInfo().WeakLevel:
+	// 	a.HandleBadTreaty(msg)
+	// 	return
+	// case a.clashOfTreaties(treaty):
+	// 	a.HandleBadTreaty(msg)
+	// 	return
+	// }
+
+	if ((treaty.ConditionOp() == messages.LE || treaty.ConditionOp() == messages.LT) && treaty.Condition() == messages.HP || treaty.Condition() == messages.AvailableFood) ||
+		((treaty.ConditionOp() == messages.GE || treaty.ConditionOp() == messages.GT) && treaty.Condition() == messages.Floor) ||
+		(treaty.Request() == messages.Inform) ||
+		((treaty.Request() == messages.LeaveAmountFood || treaty.Request() == messages.LeavePercentFood && treaty.RequestOp() == messages.LT) || treaty.RequestOp() == messages.LE) ||
+		(treaty.Condition() == messages.HP && treaty.ConditionValue() < a.HealthInfo().WeakLevel) ||
+		(a.clashOfTreaties(treaty)) {
+		a.HandleBadTreaty(msg)
+		return
+	}
```

Comment:
The switch statement is a lot better than the six line if statement - you can use the `fallthrough` keyword then you only need to call `a.HandleBadTreaty(msg)` once.

Reply:
Example: https://github.com/SOMAS2021/SOMAS2021/blob/main/pkg/agents/team5/treaty.go#L34